crates/lib3h/src/transport/websocket/actor.rs

```diff
@@ -457,76 +459,28 @@ mod tests {
             .expect("exists")
             .as_context_endpoint_builder()
             .request_id_prefix("twss_to_child2")
-            .build::<()>();
+            .build::<Option<String>>();
 
         // create two memory bindings so that we have addresses
         assert_eq!(transport1.bound_url, None);
         assert_eq!(transport2.bound_url, None);
 
-        let init_transport1_address: Lib3hUri = Url::parse(&format!("wss://127.0.0.1:{}", 1024))
-            .unwrap()
-            .into();
-
-        // closure for the wait1_for_repeatable_callback! that builds a new request with a different port number
-        // an a modified regular expression expectation to match
-        let request_fn = Box::new(|transport1_address: Lib3hUri| {
-            let old_port = transport1_address.port().unwrap_or_else(|| 0);
-            let port = get_available_port(old_port + 1).expect("Must be able to find free port");
-            let expected_transport1_address: Lib3hUri =
-                Url::parse(&format!("wss://127.0.0.1:{}", port))
-                    .unwrap()
-                    .into();
-
-            let request = RequestToChild::Bind {
-                spec: expected_transport1_address.clone(),
-            };
-            let re = format!(
-                "Response\\(Ok\\(Bind\\(BindResultData \\{{ bound_url: Lib3hUri\\(\"wss://127.0.0.1:{}/\"\\) \\}}\\)\\)\\)",
-                port.clone(),
-            );
-
-            (request.clone(), re, expected_transport1_address.clone())
-        });
-
-        let (_is_match, expected_transport1_address) = wait1_for_repeatable_callback!(
-            transport1,
-            t1_endpoint,
-            request_fn,
-            init_transport1_address
-        );
+        let init_transport1_address: Lib3hUri =
+            lib3h_protocol::uri::Builder::with_raw_url("wss://127.0.0.1/")
+                .unwrap()
+                .with_port(1024)
+                .build();
+
+        let (_is_match, expected_transport1_address) =
+            wait_for_bind_result!(transport1, t1_endpoint, init_transport1_address);
 
         let port2 = get_available_port(expected_transport1_address.port().unwrap_or_else(|| 0))
```

Comment:
do we want this `unwrap_or_else(|| 0)` here? or should it just fail if it can't find one?

Reply:
(alternately, should it default to 1025 or something instead of 0?)

Reply:
@neonphog I thought zero was good choice so it dynamically picks something but open to any behavior you think makes sense.